src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1368,30 +1372,30 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly customDriverAnnotationReport = remoteData<{ hasBinary:boolean, tiers:string[] }>({
+        await:()=>[
+            this.mutations
+        ],
+        invoke:()=>{
+            return Promise.resolve(computeCustomDriverAnnotationReport(this.mutations.result!));
+        },
+        onResult:result=>{
+            initializeCustomDriverAnnotationSettings(
+                result!,
+                this.mutationAnnotationSettings,
+                !!AppConfig.oncoprintCustomDriverTiersAnnotationDefault,
+                AppConfig.oncoprintOncoKbHotspotsDefault === "custom"
+            );
+        }
+    });
+
     readonly putativeDriverAnnotatedMutations = remoteData<AnnotatedMutation[]>({
         await:()=>[
             this.mutations,
             this.getPutativeDriverInfo
         ],
         invoke:()=>{
-            return Promise.resolve(this.mutations.result!.reduce((annotated:AnnotatedMutation[], mutation:Mutation)=>{
-                // annotate
-                const putativeDriverInfo = this.getPutativeDriverInfo.result!(mutation);
-                const putativeDriver =
-                    !!(putativeDriverInfo.oncoKb ||
-                    putativeDriverInfo.hotspots ||
-                    putativeDriverInfo.cbioportalCount ||
-                    putativeDriverInfo.cosmicCount);
-                if (putativeDriver || !this.mutationAnnotationSettings.ignoreUnknown) {
-                    annotated.push(Object.assign({
-                        putativeDriver,
-                        isHotspot: putativeDriverInfo.hotspots,
-                        oncoKbOncogenic: putativeDriverInfo.oncoKb,
-                        simplifiedMutationType: getSimplifiedMutationType(mutation.mutationType)
-                    }, mutation) as AnnotatedMutation);
-                }
-                return annotated;
-            }, []));
+            return Promise.resolve(computePutativeDriverAnnotatedMutations(this.mutations.result!, this.getPutativeDriverInfo.result!, this.mutationAnnotationSettings.ignoreUnknown));
```

Comment:
@adamabeshouse Typescript complains about `this.mutationAnnotationSettings.ignoreUnknown`: 

```
Argument of type 'boolean | undefined' is not assignable to parameter of type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'.
```

Reply:
good catch, simple fix